src/main/java/org/opensearch/ad/settings/AnomalyDetectorSettings.java

```diff
@@ -180,6 +180,9 @@ private AnomalyDetectorSettings() {}
 
     public static final Duration HOURLY_MAINTENANCE = Duration.ofHours(1);
 
+    // saving checkpoint every 6 hours.
+    public static final Duration CHECKPOINT_SAVING_FREQ = Duration.ofHours(6);
```

Comment:
Is this the final decided value? I remember in a separate discussion that every 12 hours was a possibility.

Based on the 1 million entities experiments you've done, does 6 allow for the majority of models to be saved? Obviously the smaller the number the better, to allow for models to be saved more frequently.

Reply:
Also, should this be exposed as a public setting for customers to adjust?

Reply:
yeah, will change to 12 hours in the new commit. 6 hours may or may not be enough. I changed to expose it as a public setting so that users can adjust based on their needs.